jwst/assign_wcs/nirspec.py

```diff
@@ -358,16 +375,23 @@ def get_open_slits(input_model, reference_files=None, slit_y_range=[-.55, .55]):
     """Return the opened slits/shutters in a MOS or Fixed Slits exposure.
     """
     exp_type = input_model.meta.exposure.type.lower()
-    if exp_type in ["nrs_msaspec", "nrs_autoflat"]:
+    lamp_mode = input_model.meta.instrument.lamp_mode
+    if type(lamp_mode) == str:
+        lamp_mode = lamp_mode.lower()
+    else:
```

Comment:
Is this just to guard against input files that don't have the OP_MODE keyword?

Reply:
Yes, I put this code in before I changed datamodels to set a default value for lamp_mode, it was barfing against trying to change a NoneType to lower().  It shouldn't be needed now.